src/main/java/frc/robot/utilities/pid/TrigonPIDFController.java

```diff
@@ -0,0 +1,85 @@
+package frc.robot.utilities.pid;
+
+import edu.wpi.first.wpilibj.controller.PIDController;
+
+public class TrigonPIDFController extends PIDController implements PIDFConfigurable {
+    private double v;
+    private double s;
+    private boolean isTuning;
+    private PIDFCoefs pidfCoefs;
+
+    public TrigonPIDFController(PIDFCoefs pidfCoefs) {
+        super(pidfCoefs.getKP(), pidfCoefs.getKI(), pidfCoefs.getKD());
+        this.pidfCoefs = pidfCoefs;
+    }
+
+    /**
+     * Returns the next output of the PID controller using Feed-Forward.
+     *
+     * @param measurement The current measurement of the process variable.
+     */
+    @Override
+    public double calculate(double measurement) {
+        return super.calculate(measurement) + v * getSetpoint() + s * Math.signum(getSetpoint());
```

Comment:
by `v * getSetpoint()` you assume that this is a controller for a velocity and not a position

Reply:
well yes and no. 
the pid doesn't run on somthing else as the feedforward. if you are doing ProfiledPID which doesn't exist yet on this branch you will need to run a PIDF on the position and PIDF on velocity

Reply:
It's like making a function to add numbers, that only works when you give it numbers that start with the digit 4. Either specify this limitation in the class name or solve it.